scripts/kubestellar

```diff
@@ -0,0 +1,221 @@
+        then { log_folder="$2"; shift; }
+        else { echo "$0: missing log folder" >&2; exit 1; }
+        fi;;
+    (--verbose|-V)
+        verbosity=1;;
+    (-h|--help)
+        echo "Usage: $0 [start| stop][--log-folder log_folder] [-V|--verbose]"
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2 ; exit 1;
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2; exit 1;
+        exit 1;;
+    esac
+    shift
+done
+
+
+# Check if docker is running
+if ! docker ps > /dev/null
+then
+  echo "Docker Not running ...."
+  exit 1
+fi
+
+# Check go version
+go_version=`go version | { read _ _ v _; echo ${v#go}; }`
+
+function ver { printf "%03d%03d%03d%03d" $(echo "$1" | tr '.' ' '); }
+if [ $(ver $go_version) -lt $(ver 1.19) ]; then
+    echo "Update your go version to at least 1.19"
+fi
+
+# Check if a given process name is running
+process_running() {
+  SERVICE="$1"
+  if pgrep -f "$SERVICE" >/dev/null
+  then
+      echo "running"
+  else
+      echo "stopped" 
+  fi
+}
+
+# Check if kcp is running
+if [ $(process_running kcp) != "running" ]
+then
+    echo "kcp is not running - please start it ...."
+    exit 1
+fi
+
+# Check mailbox-controller is already running
+if [ $(process_running mailbox-controller) == "running" ]
+then
+    echo "An older deployment of mailbox-controller is already running - terminating it ...."
+    pkill -f mailbox-controller
+fi
+
+# Check kubestellar-scheduler is already running
+if [ $(process_running "scheduler -v 2") == "running" ]
+then
+    echo "An older deployment of kubestellar-scheduler is already running - terminating it ...."
+    pkill -f "scheduler -v 2"
+fi
+
+# Check placement-translator is already running
+if [ $(process_running placement-translator) == "running" ]
+then
+    echo "An older deployment of placement-translator is already running - terminating it ...."
+    pkill -f placement-translator
+fi
+
+kubectl ws root &> /dev/null
+
+if [ $install -eq 0 ]; then 
+   if kubectl get Workspace "$espw_name" &> /dev/null; then
+      kubectl delete ws "$espw_name"
+      echo "Deleted workspace: $espw_name"
+   fi
+   echo "kubestellar stopped ....."
+   exit 0
+fi
+
+
+wait_for_process(){
+  status=$(process_running $1)
+  MAX_RETRIES=5
+  retries=0
+  status_code=0
+  while [ $status != "running" ]; do
+      if [ $retries -eq $MAX_RETRIES ]; then
+           status_code=1
+           break
+      fi
+
+      retries=$(( retries + 1 ))
+      sleep 3
+      status=$(process_running $1)
+  done
+  echo $status_code
+}
+
+
+# Start the kubestellar controllers
+echo "****************************************"
+echo "Launching KubeStellar ..."
+echo "****************************************"
+
+
+if kubectl get Workspace "$espw_name" &> /dev/null; then
+   echo "espw workspace already exists -- using it:"
+   kubectl ws "$espw_name"
+else 
+   if [ $verbosity == 1 ]; then
+        kubectl ws create "$espw_name" --enter
+        kubectl apply -f  ./config/crds 
```

Comment:
`kubectl apply -f  ./config/crds` is unnecessary and, I suspect, harmful.  This script should not do that.

Reply:
why is this harmful?  How else do we install the CRDs needed to run our example(s)?  @MikeSpreitzer

Reply:
need this explained so that we can resolve it ASAP @MikeSpreitzer